src/test/kotlin/no/nav/tilleggsstonader/sak/behandlingsflyt/BehandlingFlytTest.kt

```diff
@@ -0,0 +1,185 @@
+
+    private fun godkjennTotrinnskontroll(behandlingId: UUID) {
+        totrinnskontrollController.beslutteVedtak(behandlingId, BeslutteVedtakDto(true))
+    }
+
+    private fun underkjennTotrinnskontroll(behandlingId: UUID) {
+        totrinnskontrollController.beslutteVedtak(
+            behandlingId,
+            BeslutteVedtakDto(false, "", listOf(ÅrsakUnderkjent.VEDTAK_OG_BEREGNING)),
+        )
+    }
+
+    private fun opprettBehandlingOgSendTilBeslutter(personIdent: String): UUID {
+        val behandlingId = opprettTestBehandlingController.opprettBehandling(TestBehandlingRequest(personIdent))
+        opprettOppgave(behandlingId)
+        vilkårController.getVilkår(behandlingId)
+        testSaksbehandlingController.utfyllVilkår(behandlingId)
+
+        opprettVedtak(behandlingId)
+        genererSaksbehandlerBrev(behandlingId)
+        sendTilBeslutter(behandlingId)
+        return behandlingId
+    }
+
+    private fun sendTilBeslutter(behandlingId: UUID) {
+        totrinnskontrollController.sendTilBeslutter(behandlingId)
+    }
+
+    private fun verifiserBehandlingIverksettes(behandlingId: UUID) {
+        with(testoppsettService.hentBehandling(behandlingId)) {
+            assertThat(status).isEqualTo(BehandlingStatus.IVERKSETTER_VEDTAK)
+            assertThat(steg).isEqualTo(StegType.VENTE_PÅ_STATUS_FRA_UTBETALING)
+        }
+    }
+
+    private fun assertStatusTotrinnskontroll(behandlingId: UUID, expectedStatus: TotrinnkontrollStatus) {
+        with(totrinnskontrollController.hentTotrinnskontroll(behandlingId)) {
+            assertThat(status).isEqualTo(expectedStatus)
+        }
+    }
+
+    private fun opprettOppgave(behandlingId: UUID, type: Oppgavetype = Oppgavetype.BehandleSak) {
+        val oppgaveId = Random.nextInt(100, 200).toLong()
+        val oppgave = OppgaveDomain(
+            gsakOppgaveId = oppgaveId,
+            behandlingId = behandlingId,
+            type = type,
+        )
+        oppgaveRepository.insert(oppgave)
+    }
+
+    private fun genererSaksbehandlerBrev(behandlingId: UUID) {
+        val html = "SAKSBEHANDLER_SIGNATUR, BREVDATO_PLACEHOLDER, BESLUTTER_SIGNATUR"
+        brevController.genererPdf(GenererPdfRequest(html), behandlingId)
+    }
+
+    private fun opprettVedtak(behandlingId: UUID) {
+        val barn = barnService.finnBarnPåBehandling(behandlingId)
+        val utgifter = barn.first().let { mapOf(it.id to listOf(utgift())) }
+        val stønadsperioder = listOf(stønadsperiode())
+        tilsynBarnVedtakController.lagreVedtak(
+            behandlingId,
+            InnvilgelseTilsynBarnDto(stønadsperioder, utgifter),
+        )
+    }
+
+    private fun stønadsperiode() = Stønadsperiode(
+        fom = LocalDate.of(2023, 1, 1),
+        tom = LocalDate.of(2023, 1, 31),
+    )
+
+    private fun utgift() = Utgift(
+        fom = YearMonth.of(2023, 1),
+        tom = YearMonth.of(2023, 1),
+        utgift = 1000,
+    )
+
+    private fun <T> saksbehandler(fn: () -> T) =
```

Comment:
Bare et forslag, men synes det blir bittelitt enklere å se hva som skjer uten å se på funksjonen 🤓
```suggestion
    private fun <T> somSaksbehandler(fn: () -> T) =
```